docs/contributing/coding-guidelines/ros-nodes/message-guidelines.md

```diff
@@ -0,0 +1,137 @@
+| distance | millimeter       | `_mm`   |
+| distance | kilometer        | `_km`   |
+| angle    | degree (deg)     | `_deg`  |
+| time     | nanosecond       | `_ns`   |
+| time     | microsecond      | `_us`   |
+| time     | millisecond      | `_ms`   |
+| time     | minute           | `_min`  |
+| time     | hour (h)         | `_hour` |
+| velocity | km/h             | `_kmph` |
+
+## Message field types
+
+For list of types supported by the ROS interfaces [see here](https://docs.ros.org/en/galactic/Concepts/About-ROS-Interfaces.html#field-types).
+
+Also copied here for convenience:
+
+| Message Field Type | C++ equivalent   |
+| ------------------ | ---------------- |
+| `bool`             | `bool`           |
+| `byte`             | `uint8_t`        |
+| `char`             | `char`           |
+| `float32`          | `float`          |
+| `float64`          | `double`         |
+| `int8`             | `int8_t`         |
+| `uint8`            | `uint8_t`        |
+| `int16`            | `int16_t`        |
+| `uint16`           | `uint16_t`       |
+| `int32`            | `int32_t`        |
+| `uint32`           | `uint32_t`       |
+| `int64`            | `int64_t`        |
+| `uint64`           | `uint64_t`       |
+| `string`           | `std::string`    |
+| `wstring`          | `std::u16string` |
+
+## Enumerations
+
+ROS2 interfaces don't support enumerations directly.
+
+It is possible to define integers constants and assign them to a non-constant integer parameter.
+
+!!! success ""
+
+    Constants are written in `CONSTANT_CASE`.
+
+!!! success ""
+
+    Assign a different value to each element of a constant.
+
+Example from [shape_msgs/msg/SolidPrimitive.msg](https://github.com/ros2/common_interfaces/blob/f3cb4848560e91596e7688e8ac1816828fa460cb/shape_msgs/msg/SolidPrimitive.msg#L4-L11)
+
+```text
+uint8 BOX=1
+uint8 SPHERE=2
+uint8 CYLINDER=3
+uint8 CONE=4
+uint8 PRISM=5
+
+# The type of the shape
+uint8 type
+```
+
+## Comments
+
+On top of the message, briefly explain what the message contains and/or what it is used for. See [sensor_msgs/msg/Imu.msg](https://github.com/ros2/common_interfaces/blob/master/sensor_msgs/msg/Imu.msg#L1-L13) for an example.
+
+!!! tip ""
+
+    If necessary, add line comments before the fields that explain the context and/or meaning.
+
+!!! tip ""
+
+    For simple fields like `x, y, z, w` you might not need to add comments.
+
+!!! success ""
+
+    Even though it is not strictly checked, try not to pass 100 characters in a line.
+
+_Example:_
+
+```text
+# Number of times the vehicle performed an emergency brake
+uint32 count_emergency_brake
+
+# Seconds passed since the last emergency brake
+uint64 duration
+```
+
+## Example usages
+
+- Don't use unit suffixes for default types:
+  - Bad: `float32 path_length_m`
+  - Good: `float32 path_length`
+- Don't prefix the units:
+  - Bad: `float32 kmph_velocity_vehicle`
+  - Good: `float32 velocity_vehicle_kmph`
+- Use recommended suffixes if available:
```

Comment:
[Nits] `if available` is a bit unclear to me.
Could you clarify this? Is it different from `for non-default types`?

Reply:
It is for non-default types, yes. Use the recommended ones, not something made up. 

Example: for denoting `km/h`, use `kmph`, not `km_h`

Reply:
If available in the table.

Reply:
So how about writing like "if available on the table of `Non-default units`", or rephrase it?

Also, if you use `if available`, I feel we should write the case of `if not available`. I guess it would "Please confirm the maintainers".

What do you think about it?

Reply:
https://github.com/autowarefoundation/autoware-documentation/pull/94/commits/1b9ae53151af38fe3af022ccd4ad9a4de8e02cd6 ?

Reply:
Yeah, thank you! It's clear.

And what do you think about this?

> Also, if you use if available, I feel we should write the case of if not available.

Reply:
hmm makes sense. I recommend them to create an issue for it here.

Reply:
https://github.com/autowarefoundation/autoware-documentation/pull/94/commits/4b524c6998dc36a8658982fc5c24641cb2d926d5 do you think this is ok?

Reply:
Yes, perfect!